pandas/tests/test_groupby.py

```diff
@@ -3402,6 +3402,46 @@ def test_groupby_categorical_no_compress(self):
         exp = np.array([1,2,4,np.nan])
         self.assert_numpy_array_equivalent(result, exp)
 
+    def test_groupby_first_integers(self):
```

Comment:
Some of these tests have a suspiciously similar logic. Can you use loops and/or setup methods to consolidate them?